Boton para compartir tonto del dia

JLeonN opened the pull request.

JLeonN committed:
Boton whatsapp

JLeonN committed:
Boton para compartir

francosang commented on frontend/src/Componentes/Cpmpartir.jsx:
```diff
@@ -0,0 +1,18 @@
+import { Link } from "react-router-dom";
+
+const Cpmpartir = () => {
```
Esto no se usa?
Borralo 🪚

JLeonN replied:
Me olvide de borrar

francosang commented on frontend/src/Pagina/ElMasTonto/PerfilDeParticipante/EstructuraDePerfil.jsx:
```diff
@@ -19,6 +19,7 @@ const EstructuraDePerfil = () => {
         nombre={participante.nombre}
         total={participante.total}
         titulos={participante.titulos}
+        mensaje="Tonto seleccionado "
```
Porque este dice "Tonto seleccionado" y en las otras dice "El tonto del día de hoy es"?

JLeonN replied:
Porque yo puse el botón dentro de la tarjeta, y eso hace que donde se muestre la tarjeta, se va a mostrar el botón. Y cada vez que mostramos la tarjeta, no siempre es para el tonto del día; también puede ser en el perfil.

francosang commented on frontend/src/Pagina/Inicio.jsx:
```diff
@@ -33,6 +33,7 @@ const Inicio = () => {
+            mensaje="Tonto del día de hoy es "
           />
         )}
         {errorTonto && (
```
Que pasa si hay error y se muestra la tarjeta de procesando?

JLeonN replied:
No sé jaja, esa es una pregunta que tengo desde anoche.
También tengo otras preguntas.

JLeonN committed:
Borrando componente que no se usa.

JLeonN committed:
Posible error

JLeonN committed:
Posible error en cowboy

JLeonN committed:
Mensaje para Whatsapp

francosang commented on frontend/src/Componentes/Tarjeta/Tarjeta.jsx:
```diff
@@ -1,4 +1,21 @@
-const TarjetaTonta = ({ nombre, total, titulos }) => {
+import { Link } from "react-router-dom";
+
+const TarjetaTonta = ({ nombre, total, titulos, mensaje, todaLaInfo }) => {
```
Creo que hay dos variables para resolver un mismo problema.
Esto se podria resolver con una sola variable.

Te explico mas adelante.

francosang replied:
Hay dos opciones para mejorar esto: borrar la variable `todaLaInfo` o borrar la variable `mensaje`.
Pero una de las variables se tiene que ir. 
No se pueden quedar las dos porque es confuso y no queda claro.

francosang replied:
# Opcion 1: Borrar `todaLaInfo`

Si se borra la variable `todaLaInfo`, el componente Tarjeta ya no va a ser responsable de decidir si arma el mensaje del dia o el mensaje con toda la info.

Como solo quedaria la variable `mensaje`, seria la responsabilidad de cada pantalla construir el mensaje que quiere mostrar en la tarjeta.

# Opcion 2: Borrar `mensaje`

Si se borra la variable `mensaje`, el componente tarjeta va a tener toda la responsabilidad de armar el mensaje.

Dependiendo de `todaLaInfo`, el componente Tarjeta va a armar el mensaje del dia o el mensaje de toda la info.

Como la variable `mensaje` ya no existiria, las pantallas que usen la Tarjeta no van a poder mandar un valor de mensaje, solo van a poder decidir si mostrar toda la info o no.

francosang commented on frontend/src/Pagina/CowboyDelDia/CowboyDelDia.jsx:
```diff
@@ -29,6 +29,8 @@ const CowboyDelDia = () => {
           nombre={cowboy.name}
           total={"--"}
           titulos={["--Próximamente--"]}
+          mensaje="Cowboy del día de hoy es "
+          todaLaInfo=""
```
Estas pasando dos variables:
 - `mensaje`: con el texto que va a estar en el mensaje
 - `todaLaInfo`: que la usas solo para controlar que tipo de mensaje mandas, si es nulo mandas el mensaje chiquito y si es "" mandas el mensaje grande.

Creo que se podria usar una sola varaiable.
Hay que decidir si usas solo la variable `mensaje` o solo la variable `todaLaInfo`.

francosang commented on frontend/src/Pagina/ElMasTonto/PerfilDeParticipante/EstructuraDePerfil.jsx:
```diff
@@ -19,6 +19,8 @@ const EstructuraDePerfil = () => {
         total={participante.total}
         titulos={participante.titulos}
+        mensaje="Tonto seleccionado: "
+        todaLaInfo={participante.total + participante.titulos}
```
No tiene sentido ponerle valores a variable `todaLaInfo` porque el componente Tarjeta nunca usa el valor de esta variable.

La tarjeta solo compara si `todaLaInfo` es igual a `""`.

En todo caso, si se queda la variable `todaLaInfo` deberia ser booleana.

JLeonN committed:
Un margin CSS

Borré un <div> de más que había. También le di un poco de margin-top: 20px.

JLeonN committed:
Botón para compartir mejorado

francosang commented on frontend/src/Componentes/Tarjeta/Tarjeta.jsx:
```diff
@@ -1,4 +1,31 @@
+
+const TarjetaTonta = ({ nombre, total, titulos, todaLaInfo }) => {
+  let laInfo = "";
+  if (todaLaInfo == "perfil") {
+    laInfo =
+      "Tonto seleccionado: " +
+      nombre +
+      "%0A" +
+      "Veces tonto: " +
+      total +
+      "%0A" +
+      "Titulos: " +
+      titulos;
+  }
+  if (todaLaInfo == "tontoDelDía") {
+    laInfo = "Tonto del día de hoy es " + nombre;
+  }
+  if (todaLaInfo == "CowboyDelDía") {
+    laInfo = "Cowboy del día de hoy es " + nombre;
+  }
+  if (todaLaInfo == "errorTonto") {
+    laInfo = "El algoritmo está preparando un tontos " + nombre;
+  }
+  if (todaLaInfo == "errorCowboy") {
+    laInfo = "El algoritmo está preparando un cowboy " + nombre;
+  }
```
Esaaaaa! Me gusta! La variable ahora controla el tipo de mensaje.

Habría que cambiarle el nombre, a algo como:
 - `tipoDeMensaje`
 - `mensajeTipo`
 - o `tipo`
 - o algo asi
 
Que te parece?

JLeonN replied:
Si es verdad

francosang commented on frontend/src/Componentes/Tarjeta/Tarjeta.jsx:
```diff
@@ -1,4 +1,31 @@
+  if (todaLaInfo == "tontoDelDía") {
+    laInfo = "Tonto del día de hoy es " + nombre;
+  }
+  if (todaLaInfo == "CowboyDelDía") {
```
Estaba viendo que los otros valores, como `perfil` o `tontoDelDia` o `errorTonto`, empiezan con minuscula.
Pero `CowboyDelDia` empieza con mayuscula.

Mejor cambiarlo a `cowboyDelDia`.
Te parece bien?

JLeonN replied:
jaja
si claro 
no me di cuenta

JLeonN committed:
Cambiando nombres a variables.

Estoy cambiando los nombres de algunas variables para que el código sea más legible y fácil de entender.  💻

francosang commented:
Vamo @JLeonN ! Gracias por el laburo booo!

JLeonN committed:
Merge branch 'main' into Boton-para-compartir-tonto-del-dia
